### ISSUE TITLE
Typescript support?

### ISSUE THREAD

**Issue by OmgImAlexis** [therealparmesh/snoop#1]
I installed `snoop ts-node uvu typescript`, added the readme contents to `index.ts` and ran `./node_modules/.bin/ts-node index.ts`.

Seems the signatures are different.

```
➜  playground ./node_modules/.bin/ts-node index.ts

/Users/xo/code/OmgImAlexis/playground/node_modules/ts-node/src/index.ts:434
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
index.ts:26:32 - error TS2345: Argument of type '{ fn: (a: any, b: any) => any; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<(a: any, b: any) => any>[]; firstCall: Call<(a: any, b: any) => any>; ... 4 more ...; calledImmediatelyAfter: (other: <T extends (...args: any) => any>(fn: T) => { ...; }) => boolea...' is not assignable to parameter of type '<T extends (...args: any) => any>(fn: T) => { fn: T; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<T>[]; firstCall: Call<T>; lastCall: Call<...>; calledBefore: (other: typeof snoop) => boolean; calledAfter: (other: typeof snoop) => boolean; calledImmediately...'.
  Type '{ fn: (a: any, b: any) => any; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<(a: any, b: any) => any>[]; firstCall: Call<(a: any, b: any) => any>; ... 4 more ...; calledImmediatelyAfter: (other: <T extends (...args: any) => any>(fn: T) => { ...; }) => boolea...' provides no match for the signature '<T extends (...args: any) => any>(fn: T): { fn: T; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<T>[]; firstCall: Call<T>; lastCall: Call<...>; calledBefore: (other: <T extends (...args: any) => any>(fn: T) => ...) => boolean; calledAfter: (other: <T extends (...args: any) => any>(fn: T) => ...) => boolean; calledImmediatelyBefore: (other: <T extends (...args: any) => any>(fn: T) => ...) => boolean; calledImmediatelyAfter: (other: <T extends (...args: any) => any>(fn: T) => ...) => boolean; }'.

26   assert.ok(addFn.calledBefore(anotherAddFn));
                                  ~~~~~~~~~~~~
index.ts:27:31 - error TS2345: Argument of type '{ fn: (a: any, b: any) => any; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<(a: any, b: any) => any>[]; firstCall: Call<(a: any, b: any) => any>; ... 4 more ...; calledImmediatelyAfter: (other: <T extends (...args: any) => any>(fn: T) => { ...; }) => boolea...' is not assignable to parameter of type '<T extends (...args: any) => any>(fn: T) => { fn: T; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<T>[]; firstCall: Call<T>; lastCall: Call<...>; calledBefore: (other: typeof snoop) => boolean; calledAfter: (other: typeof snoop) => boolean; calledImmediately...'.

27   assert.ok(addFn.calledAfter(anotherAddFn));
                                 ~~~~~~~~~~~~
index.ts:28:43 - error TS2345: Argument of type '{ fn: (a: any, b: any) => any; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<(a: any, b: any) => any>[]; firstCall: Call<(a: any, b: any) => any>; ... 4 more ...; calledImmediatelyAfter: (other: <T extends (...args: any) => any>(fn: T) => { ...; }) => boolea...' is not assignable to parameter of type '<T extends (...args: any) => any>(fn: T) => { fn: T; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<T>[]; firstCall: Call<T>; lastCall: Call<...>; calledBefore: (other: typeof snoop) => boolean; calledAfter: (other: typeof snoop) => boolean; calledImmediately...'.

28   assert.ok(addFn.calledImmediatelyBefore(anotherAddFn));
                                             ~~~~~~~~~~~~
index.ts:29:43 - error TS2345: Argument of type '{ fn: (a: any, b: any) => any; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<(a: any, b: any) => any>[]; firstCall: Call<(a: any, b: any) => any>; ... 4 more ...; calledImmediatelyAfter: (other: <T extends (...args: any) => any>(fn: T) => { ...; }) => boolea...' is not assignable to parameter of type '<T extends (...args: any) => any>(fn: T) => { fn: T; testID: number; called: boolean; notCalled: boolean; calledOnce: boolean; callCount: number; calls: Call<T>[]; firstCall: Call<T>; lastCall: Call<...>; calledBefore: (other: typeof snoop) => boolean; calledAfter: (other: typeof snoop) => boolean; calledImmediately...'.

29   assert.not(addFn.calledImmediatelyAfter(anotherAddFn));
                                             ~~~~~~~~~~~~

    at createTSError (/Users/xo/code/OmgImAlexis/playground/node_modules/ts-node/src/index.ts:434:12)
    at reportTSError (/Users/xo/code/OmgImAlexis/playground/node_modules/ts-node/src/index.ts:438:19)
    at getOutput (/Users/xo/code/OmgImAlexis/playground/node_modules/ts-node/src/index.ts:578:36)
    at Object.compile (/Users/xo/code/OmgImAlexis/playground/node_modules/ts-node/src/index.ts:775:32)
    at Module.m._compile (/Users/xo/code/OmgImAlexis/playground/node_modules/ts-node/src/index.ts:858:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:1138:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/xo/code/OmgImAlexis/playground/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:982:32)
    at Function.Module._load (internal/modules/cjs/loader.js:875:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
```

**Comment by therealparmesh** [comment]
I see what the issue is. Thank you for reporting this!

**Comment by therealparmesh** [comment]
I believe I fixed this in 1.0.1.

**Comment by OmgImAlexis** [comment]
That's better. ^_^

```
➜  playground ./node_modules/.bin/ts-node index.ts
•   (1 / 1)

  Total:     1
  Passed:    1
  Skipped:   0
  Duration:  1.58ms
```
